### ISSUE TITLE
Comparison functions raise exceptions

### ISSUE THREAD

**Issue by AndreasBackx** [sdispater/pendulum#52]
```python
>>> Pendulum.now() == ''
...
ValueError: String does not contain a date.
>>> Pendulum.now() == 'aa'
...
ValueError: Unknown string format
```

This is also the case when checking if a `Pendulum` object is in a list:

```python
>>> Pendulum.now() in ['']
...
ValueError: String does not contain a date.
>>> '' in [Pendulum.now()]
...
ValueError: String does not contain a date.
>>> Pendulum.now() in ['aa']
...
ValueError: Unknown string format
>>> 'aa' in [Pendulum.now()]
...
ValueError: Unknown string format
```

This _prohibits_ the use of `Pendulum` objects in collections. A possible solution is to catch ValueErrors in `_get_datetime` when the instance is a string. You have a better understanding of the internals of Pendulum than I do, perhaps you think it's a better idea to catch exceptions in the comparison methods and return `False` in case of an exception.

**Comment by sdispater** [comment]
Thanks for reporting this.

There are two sides to this issue:

- First, the possible comparison between `Pendulum` instances and strings or integer. This is an idea that will create problems more than it will solve any.
- Second, comparison methods do not catch `ValueError` exceptions.

Anyway, the best approach for this is to catch the `ValueError` exceptions and return `NotImplemented` since if an exception is raised it means that `other` is not compatible.

I will push a fix and it should land officially in the next bugfix release next week.

**Comment by AndreasBackx** [comment]
Good to hear. I feel ashamed for these 3 references btw. Close your eyes.

**Comment by sdispater** [comment]
Don't worry :-)

The idea was there anyway and it would have worked, so no need to feel ashamed.

**Comment by AndreasBackx** [comment]
Commit 1: I commited my `.python-version` file.
Commit 2: somehow it inserted 3 spaces instead of 4.

I feel like I had to say that. GitHub should make these references like they are in PRs, which I thought they were. Lesson learned I guess: "Don't reference issues in commit messages."

On a side note regarding #48, I have implemented `Period.exclude` along with the previous methods. You could always use these in case you want to implement these here. But you said you wanted to do it only for one? `Period._exclude` does it only for one period, but still returns a list of periods.

**Comment by sdispater** [comment]
I just released the `0.6.1` version that fixes the issue.
